sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -720,31 +719,36 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
   private def writeMapToStringBuilder(
       kt: DataType,
       vt: DataType,
-      map: String,
-      buffer: String,
-      ctx: CodegenContext): String = {
+      map: ExprValue,
+      buffer: ExprValue,
+      ctx: CodegenContext): Block = {
 
     def dataToStringFunc(func: String, dataType: DataType) = {
       val funcName = ctx.freshName(func)
       val dataToStringCode = castToStringCode(dataType, ctx)
+      val data = JavaCode.variable("data", dataType)
+      val dataStr = JavaCode.variable("dataStr", StringType)
       ctx.addNewFunction(funcName,
```

Comment:
nit: probably we can return this as `inline`, so we don't have to put it everywhere we use it

Reply:
Maybe we can consider this in follow-ups.

Reply:
Since this method `dataToStringFunc()` is not used in other files, it would be good to address it in this PR. WDYT?

Reply:
Ok. Sounds good. Will update later.